### PR TITLE
Close empty active window when Close command is given

### PR DESCRIPTION
**Pull request by DouweM**
On OS X, Command-W is expected to close the current window, unless the current window has tabs (Safari, Chrome, Finder), in which case it will close the current tab, until only one is left, in which case it closes the entire window (and thus also the tab).

Currently in Atom, Command-W will close every pane item in the current pane, then the current pane, and so on until every pane is empty, which makes sense since pane items and panes ~ tabs. However, when only the empty root pane is left, which the user experiences as there being no panes left at all, hitting Command-W will not close the window, but will instead have no effect at all. To actually close the window, one needs to either manually hit the red close button in the upper left corner or the window, choose "Close Window" in the File menu or use the Cmd-Shift-W shortcut, all of which break the OS X UX contract.

With this PR, Command-W will behave exactly as it does right now, with one exception: when only the root pane is left, Command-W closes the entire window.

Hitting Command-W with only one pane item ~ tab open will NOT close the window, which is inconsistent with tabbed web browsing but makes more sense for a text editor.

### CODE REVIEW

**Comment by mnquintana** [comment]
Would you mind adding specs for this? Thanks!

**Comment by DouweM** [comment]
@mnquintana I amended the existing test: https://github.com/atom/atom/pull/9607/files#diff-013fa854db95a703465103ef15c4b0edR1470. Is that enough?

**Comment by benogle** [comment]
Refs https://github.com/atom/atom/issues/863

**Comment by nathansobo** [comment]
Great job @DouweM! :zap: We'll see if everyone yells at us for changing it to work this way now. :grimacing:

**Comment by nathansobo** [comment]
Code looks good, restarting the failed Travis build.

**Comment by benogle** [comment]
IMO, there should be an option to preserve the current behavior

**Comment by benogle** [comment]
@DouweM Can you add a setting for this? Something like `closeEmptyWindowOnCoreClose`? @nathansobo feel free to come up with a better name.

**Comment by DouweM** [comment]
@benogle May I ask why? The old behavior has never made sense to me or any other Mac/Linux Atom user I know (I don't know any Windows Atom users). You'll always have people complain about changes in behavior, but I think this is a change for the better and I'm seriously wary about adding configuration options just because people don't like change.

**Comment by benogle** [comment]
For behavior changes like this, we need to give people who rely on the current behavior an out. Similar to the recent changes in fuzzy-finder matching algorithm, if it's clear that almost no one wants the old behavior, we can pull the option in the future.

> I think this is a change for the better

I dont necessarily think this is a change for the better. It's just different. I for one really love the current behavior, commands do explicitly one thing: close an item or close a window and do not change based on state. I can mash cmd-w 100 times, and know that I wont accidentally close the window. We honestly have had little to no complaining about the current behavior in the last year or so. I'm inclined to think that many people are now used to it.

**Comment by nathansobo** [comment]
Agreed on adding the option. There's no reason to have a bike shed debate on something that's so easy to customize. It originally worked this way and we changed it based on internal feedback from GitHubbers, so I think there's a good argument that it's going to be controversial. Let's avoid the controversy and let people design their own experience.

**Comment by DouweM** [comment]
@nathansobo Fair enough, I will add an option!

**Comment by DouweM** [comment]
@nathansobo Option was added!

**Comment by nathansobo** [comment]
Kicking off another Travis build.

**Comment by nathansobo** [comment]
Our Travis build is timing out. Think it may have to do with Blink throttling our timers and animation frames on CI. I'm running a build locally.

**Comment by nathansobo** [comment]
Okay, I figured it out. @DouweM locally I'm seeing the spec you added actually closing the test window, which prevents us from informing the browser process that the tests completed. Could you move the `spyOn(atom, 'close')` into your before each and see if that gets us a :green_apple: Travis build?

**Comment by nathansobo** [comment]
Actually, nevermind, I'll just move it and merge.

**Comment by nathansobo** [comment]
:metro:  Thanks for your help @DouweM. :metro:

**Comment by DouweM** [comment]
@nathansobo :heart:

File: src/config-schema.coffee [hunk 1]
```diff
@@ -21,7 +21,6 @@ module.exports =
       followSymlinks:
         type: 'boolean'
         default: true
-        title: 'Follow symlinks'
```
Comment (matiasgarciaisaia) [inline src/config-schema.coffee]: Was this deletion intentional?

Comment (DouweM) [inline src/config-schema.coffee]: It was, te default title is the key (`followSymlinks`) capitalized, which would be `Follow Symlinks`. The downcase `s` was inconsistent with the other titles.

**Pull request merged**